src/java/org/apache/cassandra/config/CassandraRelevantProperties.java

```diff
@@ -269,6 +269,12 @@
     SYSTEM_DISTRIBUTED_NTS_RF_OVERRIDE_PROPERTY("cassandra.system_distributed_replication_per_dc"),
     SYSTEM_DISTRIBUTED_NTS_DC_OVERRIDE_PROPERTY("cassandra.system_distributed_replication_dc_names"),
 
+    /** Maximum time to wait for node information persistence operations, like save, delete or sync, before throwing exception - this is many for unfound deadlock mitigation */
+    NODES_PERSISTENCE_TIMEOUT_MS("cassandra.nodes.persistence_timeout_ms", "60000"),
+
+    /** Maximum time to wait to acquire a lock for Gossiper task before failing - this is many for unfound deadlock mitigation */
+    GOSSIPER_LOCK_TIMEOUT("cassandra.nodes.lock_timeout_ms", "60000"),
```

Comment:
Is there a history in Cassandra of providing these kinds of tunables? I can see the value of having these options during testing to help with debugging, but this doesn't read like something that's restricted to testing/debugging and I fear that most users will not understand how to set this value correctly and shouldn't be allowed to.

In particular, this is the kind of option which users tend to customise in an attempt to "make things go faster".

Am I overreacting? Is there any way we can make this specific to testing?

Reply:
hmm... I just thought about not advertising those and keep them as internal settings. Do you think I should make them start with `cassandra.test.` ?

Reply:
I think the idea of prefixing them with `cassandra.test` yeah.

Reply:
I would make this a multiple of one of the already existing gossip influencing -D’s